source/read/change-streams.txt

```diff
@@ -0,0 +1,278 @@
+- ``FullDocumentBeforeChange.REQUIRED``: The change event includes a pre-image of the
+  modified document for change events. If the pre-image is not available, the
+  driver raises an error.
+
+The **post-image** is the full version of a document *after* a change. To include the
+post-image in the change stream event, pass the ``fullDocument()`` method one of the
+following parameters:
+
+- ``FullDocument.UPDATE_LOOKUP``: The change event includes a copy of the entire changed
+  document from some time after the change.
+- ``FullDocument.WHEN_AVAILABLE``: The change event includes a post-image of the
+  modified document for change events only if the post-image is available.
+- ``FullDocument.REQUIRED``: The change event includes a post-image of the
+  modified document for change events. If the post-image is not available, the
+  driver raises an error.
+
+The following example calls the ``watch()`` method on a collection and includes the post-image
+of updated documents by specifying the ``fullDocument`` parameter:
+
+.. literalinclude:: /includes/read/change-streams.kt
+   :start-after: start-change-stream-post-image
+   :end-before: end-change-stream-post-image
+   :language: kotlin
+   :copyable:
+   :dedent:
+
+With the change stream application running, updating a document in the
+``restaurants`` collection by using the :ref:`preceding update example
+<kotlin-sync-change-stream-update>` prints a change event resembling the following:
+
+
+.. code-block:: sh
+   
+   ChangeStreamDocument{ operationType=update, resumeToken={"_data": "..."},
+   namespace=sample_restaurants.restaurants, destinationNamespace=null, fullDocument=Restaurant(name=Blarney Castle, cuisine=Irish),
+   fullDocumentBeforeChange=null, documentKey={"_id": {"$oid": "..."}},
+   clusterTime=Timestamp{value=..., seconds=..., inc=...},
+   updateDescription=UpdateDescription{removedFields=[], updatedFields={"cuisine": "Irish"},
+   truncatedArrays=[], disambiguatedPaths=null}, txnNumber=null, lsid=null, splitEvent=null,
+   wallTime=BsonDateTime{value=...}}
+
+
+To learn more about pre-images and post-images, see 
+:manual:`Change Streams with Document Pre- and Post-Images </changeStreams#change-streams-with-document-pre--and-post-images>` 
+in the {+mdb-server+} manual.
+
+Additional Information
+----------------------
+
+To learn more about change streams, see :manual:`Change Streams
+</changeStreams>` in the {+mdb-server+} manual.
+
+API Documentation
+~~~~~~~~~~~~~~~~~
+
+To learn more about any of the methods or types discussed in this
+guide, see the following API documentation:
+
+- `MongoClient.watch() <{+api+}/com.mongodb.kotlin.client/-mongo-client/watch.html>`__
+- `MongoDatabase.watch() <{+api+}/com.mongodb.kotlin.client/-mongo-database/watch.html>`__
+- `MongoCollection.watch() <{+api+}/com.mongodb.kotlin.client/-mongo-collection/watch.html>`__
+- `ChangeStreamIterable <{+api+}/com.mongodb.kotlin.client/-change-stream-iterable/index.html>`__
```

Comment:
Note for reviewer: these links are broken at the moment.

Rea has a change that has yet to be merged that changes the `{+api+}` source constant. These will work once it's merged.